chain/chain/src/chain.rs

```diff
@@ -2770,6 +2783,8 @@ impl Chain {
             );
             store_update.commit()?;
             flat_storage_manager.create_flat_storage_for_shard(shard_uid).unwrap();
+            // Flat storage is ready, load memtrie if it is enabled.
```

Comment:
Is this going to execute synchronously? If so, does this happen only during startup?

If this happens not just during startup, is there a way to run this in a different thread? I don't think it is acceptable to pause the chain for 2 minutes.